backend/de.metas.contracts/src/main/java/de/metas/contracts/pricing/SubscriptionPricingRule.java

```diff
@@ -119,7 +119,8 @@ private IPricingResult aggregateSubscriptionDiscount(
 	{
 		if (subscriptionDiscountLine != null
 				&& !subscriptionPricingResult.isDisallowDiscount()
-				&& (!subscriptionPricingResult.isDiscountCalculated() || subscriptionDiscountLine.isPrioritiseOwnDiscount()))
+				&& !subscriptionDiscountLine.getDiscount().isZero()
```

Comment:
why ignore zero-discount?

Reply:
I thought in a scenario where:
- `C_SubscrDiscount_Line.Discount` = 0
- `C_SubscrDiscount_Line.IfForeignDiscountsExist` = `Use_Our_Discount`
- Non-zero discount coming in from a different rule

In this case, should we apply:
1. Always apply the discount in `C_SubscrDiscount_Line`, because the flag says so
2. Because there's not really a discount here, apply the non-zero one instead.

I found merit in both choices, and went with the option 2. Should I go with 1 instead?

Reply:
i think 1. makes more sense